src/bit-systems/mixer-animatable.ts

```diff
@@ -0,0 +1,33 @@
+import { addComponent, defineQuery, enterQuery, exitQuery, removeComponent } from "bitecs";
+import { AnimationMixer } from "three";
+import { MixerAnimatable, MixerAnimatableInitialize, MixerAnimatableData, Object3DTag } from "../bit-components";
+import { HubsWorld } from "../app";
+
+const initializeQuery = defineQuery([MixerAnimatableInitialize, Object3DTag]);
+const initializeEnterQuery = enterQuery(initializeQuery);
+const mixerQuery = defineQuery([MixerAnimatable, Object3DTag]);
+const mixerExitQuery = exitQuery(mixerQuery);
+
+export function mixerAnimatableSystem(world: HubsWorld): void {
+  initializeEnterQuery(world).forEach(eid => {
+    addComponent(world, MixerAnimatable, eid);
+
+    const object = world.eid2obj.get(eid)!;
+    const mixer = new AnimationMixer(object);
+    MixerAnimatableData.set(eid, mixer);
+
+    removeComponent(world, MixerAnimatableInitialize, eid);
+  });
```

Comment:
I have a mix feeling about it.

Pros: Yes, the implementation in model will be simpler. And the two step initializations (MixerAnimatableInitialize -> MixerAnimatable) for model will not be needed.
Cons: Object (AnimationMixer in this case) creation and clean up are spreaded out, creation is here while clean up is in mixerAnimatableSystem. I think it would be more readable and maintainable if both are placed in a single place. And MixerAnimatable initializations can be inconsistend between glTF and JSX. I think it would be more readable and maintainable if they are consistent.

Personally I prefer consistency and both creation and clean up are placed in a sinble place, so I want to go as is unless there is any critical problems.

Reply:
Yes, makes sense, I think it's better to prioritize consistency.